akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/PathDirectivesSpec.scala

```diff
@@ -350,6 +350,26 @@ class PathDirectivesSpec extends RoutingSpec with Inside {
       "accept [/1.2.3/]" inThe test("List(1, 2, 3):/")
       "accept [/1.2.3./]" inThe test("List(1, 2, 3):./")
     }
+    {
+      val test = testFor(pathPrefix(IntNumber.repeat(min = 1, max = Int.MaxValue, separator = ".")) { echoCaptureAndUnmatchedPath })
+      "reject [/baz]" inThe test()
+      "accept [/1baz]" inThe test("List(1):baz")
+      "accept [/1.baz]" inThe test("List(1):.baz")
+      "accept [/1.2baz]" inThe test("List(1, 2):baz")
+      "accept [/1.2.3.4.5.6.7]" inThe test("List(1, 2, 3, 4, 5, 6, 7):")
+    }
+    {
+      val test = testFor(pathPrefix(IntNumber.repeat(min = 0, max = 0, separator = ".")) { echoCaptureAndUnmatchedPath })
+      "accept [/qux]" inThe test("List():qux")
+      "accept [/1qux]" inThe test("List():1qux")
+    }
+    {
+      val test = testFor(pathPrefix(IntNumber.repeat(min = 0, max = 1, separator = ".")) { echoCaptureAndUnmatchedPath })
+      "accept [/xyz]" inThe test("List():xyz")
+      "accept [/1xyz]" inThe test("List(1):xyz")
+      "accept [/1.xyz]" inThe test("List(1):.xyz")
+      "accept [/1.2.xyz]" inThe test("List(1):.2.xyz")
```

Comment:
What about when it is not a prefix but the full path, would ´IntNumber.repeat(0, 1, ".")` reject for this one then? Could you check if there is a test covering that and add if not?

Reply:
I don't there is a test which matches below test content. 
```
val test = testFor(path(IntNumber.repeat(min = 0, max = 1, separator = ".")) { echoCaptureAndUnmatchedPath }) //path instead of pathPrefix

"reject [/1.2.xyz]" inThe test()
..... so on
``` 
I will add it to this PR. 

Thanks.

Reply:
Added these above tests. Thanks.

Reply:
Thanks, good to verify that it wasn't a case that broke with this fix.